package.json

```diff
@@ -5,13 +5,13 @@
   "homepage": "https://github.com/Matthew-Collins-Defra/elm-api-boilerplate",
   "main": "index.js",
   "scripts": {
+    "pretest": "npm run test:lint",
     "start": "node index.js",
-    "lint": "standard",
-    "unit-test": "lab -c -r console -o stdout -r lcov -o lcov.info",
-    "pretest": "npm run lint",
-    "test": "npm run unit-test",
-    "unit-test-watch": "nodemon --inspect=0.0.0.0 --exec npm run unit-test",
-    "snyk-test": "snyk test"
+    "test": "npm run test:unit",
+    "test:lint": "standard",
+    "test:snyk": "snyk test",
+    "test:unit": "lab -c -r console -o stdout -r junit -o ./test-output/TEST-$(date '+%Y%m%d-%H%M%S').xml",
+    "watch:unit": "nodemon --inspect=0.0.0.0 --exec npm run test:unit"
```

Comment:
Would be nice to get some pre-commit hooks in via husky or similar to run the tests before a commit.

Reply:
Interesting idea. I won't add any hooks to this PR since it goes beyond the scope but we should definitely discuss this between the teams and agree a standard.

I've known pre-commit hooks to get really annoying to the point that people just get in the habit of using the "skip hooks" argument. Basically, if you remember to run the tests more often than not then having them run again automatically is really annoying.

I realise the irony that I missed a failing test recently. Although I had actually run the tests, I didn't notice the failure because of all the coverage issues we get on every test run at the moment. We will sort those out but I'm waiting until we've switched to Jest before I pay much attention to tests.

Reply:
Pre-commit hooks aren't mentioned anywhere in Defra's proposed standards and I'm not aware of anyone using them.

However what is mentioned is pushing code to GitHub should always trigger a build.  A failed build should prevent completion of a pull request.  This is also how I've worked in the past on other projects.

As we should be following these standards I think pre-commit hooks as well would be overkill as we'd be running tests first manually as a lot of people do, then as part of the hook, then again in the pipeline.